provider/azure/internal/azurestorage/interface.go

```diff
@@ -57,5 +72,46 @@ type clientWrapper struct {
 }
 
 func (w clientWrapper) GetBlobService() BlobStorageClient {
-	return w.Client.GetBlobService()
+	return &blobStorageClient{w.Client.GetBlobService()}
+}
+
+type blobStorageClient struct {
+	storage.BlobStorageClient
+}
+
+func (c *blobStorageClient) GetContainerReference(name string) Container {
```

Comment:
should comment these pubic methods, even though comments will be a bit pithy

Reply:
done